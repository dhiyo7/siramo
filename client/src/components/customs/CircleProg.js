import React, { Component } from 'react'
import { View, Text } from 'react-native'
import * as Progress from 'react-native-progress'

class CircleProg extends Component {
  ProgText = (value, maxValue, textTitle) => {
    return `${textTitle}\n${Math.round(value * 100)}${maxValue}`
  }

  ProgConvert = (value) => {
    return value / 100
  }

  render() {
    const sensorValue = this.props.sensor
    const maxValue = this.props.maxValue
    const convertValue = this.ProgConvert(sensorValue)
    const colorValue = this.props.color
    const textTitle = this.props.textTitle
    return (
      <View style={{margin:12}}>
        <Progress.Circle 
          progress={convertValue} 
          size={120}
          showsText={true}
          formatText={() => this.ProgText(convertValue, maxValue, textTitle)}
          thickness={10}
          color={colorValue}
          spinDuration={3000}
          textStyle={{fontSize: 16, textAlign: 'center'}}
        />
      </View>
    )
  }
}

export default CircleProg