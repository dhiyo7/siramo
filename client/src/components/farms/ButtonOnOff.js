import React, { Component } from 'react'
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet
} from 'react-native'
import { inject, observer } from 'mobx-react'
import UserStore from '../../store/UserStore'

@inject('UserStore')
class ButtonOnOff extends Component {
  render() {
    const {ready_siram, triggerSiram} = this.props.siram
    let txtButton = 'Start Watering'
    let buttonStyles = styles.buttonContainerStart
    if (ready_siram === 0) {
      txtButton = 'Stop Watering'
      buttonStyles = styles.buttonContainerStop
    }
    return (
      <TouchableOpacity
        style={buttonStyles}
        onPress={triggerSiram}
      >
        <Text style={styles.buttonText}>{txtButton}</Text>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  buttonContainerStart: {
    backgroundColor: '#33691E',
    paddingVertical: 15,
    marginBottom: 4,
    borderRadius: 4
  },
  buttonContainerStop: {
    backgroundColor: '#f44242',
    paddingVertical: 15,
    marginBottom: 4,
    borderRadius: 4
  },
  buttonText: {
    color: '#fff',
    textAlign: 'center',
    fontWeight: '700'
  }
})

export default ButtonOnOff