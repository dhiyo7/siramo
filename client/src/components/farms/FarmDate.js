import React, { Component } from 'react'
import {
  View, Text, StyleSheet
} from 'react-native'
import { Ionicons } from '@expo/vector-icons'

class FarmDate extends Component {
  render() {
    const { title, dates, times } = this.props.date
    // console.log(this.props.date.dates)
    // console.log('Jam nya', times)
    // console.log('Tanggal nya', dates)
    return (
      <View>
        <Text>{title}</Text>
        <View style={styles.MainCard}>
          <View style={styles.cards}>
            <Text style={styles.cardText}>
              <Ionicons name='md-calendar' size={25} /> {dates}
            </Text>
          </View>
          <View style={styles.cards}>
            <Text style={styles.cardText}>
              <Ionicons name='md-clock' size={25} /> {times}
            </Text>
          </View>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  MainCard: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    borderWidth: 1,
    borderRadius: 2,
    borderColor: '#ddd',
    shadowColor: '#000',
    backgroundColor: '#128eb7',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.8,
    shadowRadius: 2,
    elevation: 1,
  },
  cards: {
    height: 35,
    padding: 4,
    margin: 8,
    alignItems: 'center'
  },
  cardText: {
    fontSize: 25,
    color: '#ffffff'
  }
})

export default FarmDate