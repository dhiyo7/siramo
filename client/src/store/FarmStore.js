import { observable, computed } from 'mobx'
import { db, User } from './firebase'
import { Alert } from 'react-native'


class FarmStore {
  @observable farmData = {
    historyHumidity: [],
    historyWaterRatio: [],
    historyTemperature: [],
    historyWaterLevel: [],
    historyLoading: false
  }

  @observable navigation = {}

  getHistory = (key) => {
    this.farmData.historyLoading = true
    db.ref('/history/17UFak7JqufG1RXUeVW30jwdfrQ2').on('value', (snapshot) => {

      snapshot.forEach(snap => {
        this.farmData.historyHumidity.push({
          x: snap.val().last_updated,
          y: parseFloat(snap.val().humidity)
        })
        this.farmData.historyWaterRatio.push({
          x: snap.val().last_updated,
          y: Math.round(parseFloat(snap.val().water_ratio))
        })
        this.farmData.historyTemperature.push({
          x: snap.val().last_updated,
          y: parseFloat(snap.val().temperature)
        })
        this.farmData.historyWaterLevel.push({
          x: snap.val().last_updated,
          y: parseFloat(snap.val().water_level)
        })
      })
      this.farmData.historyLoading = false
    })
  }

  setSchedule = (cronFormat, max, min) => {
    let updatedValue = {
      name: cronFormat,
      maxWaterRatio: max,
      minWaterRatio: min
    }
    console.log('Firebase Updated', updatedValue)
    db.ref(`/users`).update(updatedValue)
      .then(response => {
        console.log(response)
      })
      .catch(err => {
        console.log(err)
      })
  }

}

export default new FarmStore()