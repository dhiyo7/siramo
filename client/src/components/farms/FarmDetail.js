import React, { Component } from 'react'
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Image
} from 'react-native'
import { inject, observer } from 'mobx-react'
import UserStore from '../../store/UserStore'
import FarmStore from '../../store/FarmStore'

import CircleProg from '../customs/CircleProg'
import TemperatureGauge from '../customs/TemperatureGauge'
import { Ionicons } from '@expo/vector-icons'
import FarmPic from '../farms/FarmsPic'
import FarmDate from '../farms/FarmDate'
import ButtonOnOff from '../farms/ButtonOnOff'

@observer class FarmDetail extends Component {
  render() {
    const {
      name, temperature, water_ratio, ready_siram,
      water_level, humidity, last_siram, last_updated,
    } = FarmStore.FarmDetail
    const { dateFormat, timeFormat } = UserStore

    return (
      <View style={styles.mainContainer}>
        <FarmPic />
        <View style={styles.DetailCard}>
          <View style={styles.infoView}>
            <Text style={styles.textTitle}>{name}</Text>
          </View>
          <TemperatureGauge
            sensor={temperature}
            textTitle={'Temperature'}
            maxValue={'Â°C'}
            color='rgba(241, 6, 102, 0.83)'
          />
          <CircleProg 
            sensor={water_ratio}
            textTitle={'Water Ratio'}
            maxValue={'%'}
            color='rgba(0, 167, 5, 1)' 
          />
          <CircleProg 
            sensor={(water_level/50)*100}
            textTitle={'Water Level'}
            maxValue={'%'}
            color='rgb(41, 177, 237)' 
          />
          <CircleProg 
            sensor={humidity}
            textTitle={'Humidity'}
            maxValue={'%'}
            color='rgba(242, 129, 35, 0.83)'
          />
          <View style={styles.dateContainer}>
              <FarmDate 
                date={{
                  title: 'Last Watering:',
                  dates: dateFormat(last_siram),
                  times: timeFormat(last_siram)
                }}
              />
              <FarmDate 
                date={{
                  title: 'Last Updated:',
                  dates: dateFormat(last_updated),
                  times: timeFormat(last_updated)
                }}
              />
          </View>
          {/* <Text>Ready Siram: {ready_siram}</Text> */}
        </View>
        <ButtonOnOff siram={{ready_siram, triggerSiram: FarmStore.triggerSiram}}/>
      </View>
    )
  }
} 

const styles = StyleSheet.create({
  mainContainer: {
    alignItems: 'center',
    width: '100%',
    backgroundColor: 'white'
  },
  DetailCard: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    backgroundColor: '#fff',
    padding: 4
  },
  textTitle: {
    fontSize: 20,
    textAlign: 'center',
    width: '100%'
  },
  dateView: {
    width: '100%',
    height: 'auto',
    borderRadius: 10
  },
  dateText: {
    textAlign: 'center',
    fontSize: 20
  },
  infoView: {
    width: '100%',
  },
  infoText: {
    fontSize: 20 
  },
  dateContainer: {
    flexWrap: 'wrap',
    flexDirection: 'row',
    width: '100%',
    height: 'auto',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center'
  }
})

export default FarmDetail