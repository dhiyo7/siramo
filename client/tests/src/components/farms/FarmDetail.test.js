import React from 'react'
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Image
} from 'react-native'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import FarmStore from '../../../../src/store/FarmStore'
import userStore from '../../../../src/store/UserStore'

import FarmDetail from '../../../../src/components/farms/FarmDetail'
import CircleProg from '../../../../src/components/customs/CircleProg'
import TemperatureGauge from '../../../../src/components/customs/TemperatureGauge'
import FarmPic from '../../../../src/components/farms/FarmsPic'
import FarmDate from '../../../../src/components/farms/FarmDate'
import ButtonOnOff from '../../../../src/components/farms/ButtonOnOff'

Enzyme.configure({ adapter: new Adapter()})

let wrapper
let mockData
let mockUser

beforeEach(() => {
  FarmStore.FarmDetail.name = 'My Plant',
  FarmStore.FarmDetail.temperature = 20,
  FarmStore.FarmDetail.water_ratio = 10,
  FarmStore.FarmDetail.water_level = 30,
  FarmStore.FarmDetail.humidity = 30,
  FarmStore.FarmDetail.last_siram = 1526824574774,
  FarmStore.FarmDetail.last_updated = 1224824574774,
  FarmStore.FarmDetail.cronSchedule = '* 20 */11 * * *',
  FarmStore.FarmDetail.minWaterRatio = 63,
  FarmStore.FarmDetail.maxWaterRatio = 4,
  FarmStore.FarmDetail.ready_siram = 1,
  FarmStore.FarmDetail.loading = false

  wrapper = shallow(<FarmDetail />)
})

describe('<FarmDetail /> snapshot Testing', () => {
  it('snapshot works!', () => {
    const tree = renderer.create(<FarmDetail FarmStore={{FarmDetail:mockData}} UserStore={mockUser}/>)
    expect(tree).toMatchSnapshot()
  })
})

describe('<FarmDetail />', () => {
  it('should render <FarmDetail />', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<FarmDetail /> child is rendered', () => {
  it('should render <FarmDetail /> child Element', () => {
    expect(wrapper.find(<View />)).toBeDefined()
    // expect(wrapper.find('View').length).toBe(5)
    expect(wrapper.find(<FarmPic />)).toBeDefined()
    // expect(wrapper.find('FarmPic').length).toBe(1)
    expect(wrapper.find(<TemperatureGauge />)).toBeDefined()
    expect(wrapper.find('TemperatureGauge').length).toBe(1)
    expect(wrapper.find(<CircleProg />)).toBeDefined()
    expect(wrapper.find('CircleProg').length).toBe(3)
    expect(wrapper.find(<FarmDate />)).toBeDefined()
    expect(wrapper.find('FarmDate').length).toBe(2)
    expect(wrapper.find(<Text />)).toBeDefined()
    expect(wrapper.find('Text').length).toBe(1)
    expect(wrapper.find(<ButtonOnOff />)).toBeDefined()
    expect(wrapper.find('ButtonOnOff').length).toBe(1)
  })
})

describe('<FarmDetail /> child props is working', () => {
  it('Childern Component should have the same props value as FarmStore', () => {
    let lastWtrDate = userStore.dateFormat(FarmStore.FarmDetail.last_siram)
    let lastWtrTime = userStore.timeFormat(FarmStore.FarmDetail.last_siram)
    
    let lastUpdateDate = userStore.dateFormat(FarmStore.FarmDetail.last_updated)
    let lastUpdateTime = userStore.timeFormat(FarmStore.FarmDetail.last_updated)

    console.log(wrapper.find('ButtonOnOff').get(0))
    expect(wrapper.find('TemperatureGauge').get(0).props.sensor).toEqual(FarmStore.FarmDetail.temperature)
    expect(wrapper.find('Text').at(0).props().children).toEqual(FarmStore.FarmDetail.name)
    expect(wrapper.find('CircleProg').get(0).props.sensor).toEqual(FarmStore.FarmDetail.water_ratio)
    expect(wrapper.find('CircleProg').get(1).props.sensor).toEqual(FarmStore.FarmDetail.water_level)
    expect(wrapper.find('CircleProg').get(2).props.sensor).toEqual(FarmStore.FarmDetail.humidity)
    expect(wrapper.find('FarmDate').get(0).props.date.dates).toEqual(lastWtrDate)
    expect(wrapper.find('FarmDate').get(0).props.date.times).toEqual(lastWtrTime)
    expect(wrapper.find('FarmDate').get(1).props.date.dates).toEqual(lastUpdateDate)
    expect(wrapper.find('FarmDate').get(1).props.date.times).toEqual(lastUpdateTime)
    // button kalo On Off kalo di pencet ngubah 1 dan 0
  })
})