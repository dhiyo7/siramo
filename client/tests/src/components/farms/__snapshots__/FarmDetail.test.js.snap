// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FarmDetail /> snapshot Testing snapshot works! 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 10,
        "paddingTop": 20,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://jurnalaquascape.com/wp-content/uploads/2015/05/amazon-sword.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 75,
          "height": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F1F8E9",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "padding": 4,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        My Plant
      </Text>
    </View>
    <View
      style={
        Object {
          "margin": 12,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <ARTSurfaceView
          style={
            Array [
              Object {
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              },
              Object {
                "height": 120,
                "width": 120,
              },
            ]
          }
        >
          <ARTShape
            d={
              Array [
                0,
                60,
                6,
                3,
                89.82337649086284,
                6,
                114,
                30.176623509137148,
                114,
                59.999999999999986,
                3,
                114.00000000000003,
                89.82337649086283,
                89.82337649086287,
                114,
                60.00000000000002,
                114,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0.9450980392156862,
                0.023529411764705882,
                0.4,
                0.83,
              ]
            }
            strokeCap={0}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={10}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
          <ARTShape
            d={
              Array [
                0,
                60,
                0.5,
                4,
                60,
                60,
                59.5,
                -1.5707963267948966,
                1.5707963267948966,
                1,
                4,
                60,
                60,
                59.5,
                1.5707963267948966,
                -1.5707963267948966,
                1,
                1,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0.9450980392156862,
                0.023529411764705882,
                0.4,
                0.83,
              ]
            }
            strokeCap={0}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={1}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
        </ARTSurfaceView>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 49,
              "height": 98,
              "justifyContent": "center",
              "left": 11,
              "position": "absolute",
              "top": 11,
              "width": 98,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgba(241, 6, 102, 0.83)",
                  "fontSize": 21.77777777777778,
                  "fontWeight": "300",
                },
                Object {
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ]
            }
          >
            Temperature
20Â°C
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "margin": 12,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <ARTSurfaceView
          style={
            Array [
              Object {
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              },
              Object {
                "height": 120,
                "width": 120,
              },
            ]
          }
        >
          <ARTShape
            d={
              Array [
                0,
                60,
                6,
                3,
                71.4036797033666,
                6,
                82.5146329453613,
                9.610167552246935,
                91.74040362379355,
                16.313082303752836,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0,
                0.6549019607843137,
                0.0196078431372549,
                1,
              ]
            }
            strokeCap={0}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={10}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
          <ARTShape
            d={
              Array [
                0,
                60,
                0.5,
                4,
                60,
                60,
                59.5,
                -1.5707963267948966,
                1.5707963267948966,
                1,
                4,
                60,
                60,
                59.5,
                1.5707963267948966,
                -1.5707963267948966,
                1,
                1,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0,
                0.6549019607843137,
                0.0196078431372549,
                1,
              ]
            }
            strokeCap={0}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={1}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
        </ARTSurfaceView>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 49,
              "height": 98,
              "justifyContent": "center",
              "left": 11,
              "position": "absolute",
              "top": 11,
              "width": 98,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgba(0, 167, 5, 1)",
                  "fontSize": 21.77777777777778,
                  "fontWeight": "300",
                },
                Object {
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ]
            }
          >
            Water Ratio
10%
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "margin": 12,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <ARTSurfaceView
          style={
            Array [
              Object {
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              },
              Object {
                "height": 120,
                "width": 120,
              },
            ]
          }
        >
          <ARTShape
            d={
              Array [
                0,
                60,
                6,
                3,
                77.28567065376835,
                6,
                93.52665540997732,
                14.275195058139545,
                103.68691769624715,
                28.259596376206446,
                3,
                113.84717998251699,
                42.24399769427335,
                116.69861787112102,
                60.24726798244885,
                111.3570518799383,
                76.68691769624715,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0.1607843137254902,
                0.6941176470588235,
                0.9294117647058824,
                1,
              ]
            }
            strokeCap={0}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={10}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
          <ARTShape
            d={
              Array [
                0,
                60,
                0.5,
                4,
                60,
                60,
                59.5,
                -1.5707963267948966,
                1.5707963267948966,
                1,
                4,
                60,
                60,
                59.5,
                1.5707963267948966,
                -1.5707963267948966,
                1,
                1,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0.1607843137254902,
                0.6941176470588235,
                0.9294117647058824,
                1,
              ]
            }
            strokeCap={0}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={1}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
        </ARTSurfaceView>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 49,
              "height": 98,
              "justifyContent": "center",
              "left": 11,
              "position": "absolute",
              "top": 11,
              "width": 98,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgb(41, 177, 237)",
                  "fontSize": 21.77777777777778,
                  "fontWeight": "300",
                },
                Object {
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ]
            }
          >
            Water Level
30%
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "margin": 12,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <ARTSurfaceView
          style={
            Array [
              Object {
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              },
              Object {
                "height": 120,
                "width": 120,
              },
            ]
          }
        >
          <ARTShape
            d={
              Array [
                0,
                60,
                6,
                3,
                77.28567065376835,
                6,
                93.52665540997732,
                14.275195058139545,
                103.68691769624715,
                28.259596376206446,
                3,
                113.84717998251699,
                42.24399769427335,
                116.69861787112102,
                60.24726798244885,
                111.3570518799383,
                76.68691769624715,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0.9490196078431372,
                0.5058823529411764,
                0.13725490196078433,
                0.83,
              ]
            }
            strokeCap={0}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={10}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
          <ARTShape
            d={
              Array [
                0,
                60,
                0.5,
                4,
                60,
                60,
                59.5,
                -1.5707963267948966,
                1.5707963267948966,
                1,
                4,
                60,
                60,
                59.5,
                1.5707963267948966,
                -1.5707963267948966,
                1,
                1,
              ]
            }
            fill={null}
            opacity={1}
            stroke={
              Array [
                0.9490196078431372,
                0.5058823529411764,
                0.13725490196078433,
                0.83,
              ]
            }
            strokeCap={0}
            strokeDash={null}
            strokeJoin={1}
            strokeWidth={1}
            transform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          />
        </ARTSurfaceView>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 49,
              "height": 98,
              "justifyContent": "center",
              "left": 11,
              "position": "absolute",
              "top": 11,
              "width": 98,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgba(242, 129, 35, 0.83)",
                  "fontSize": 21.77777777777778,
                  "fontWeight": "300",
                },
                Object {
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ]
            }
          >
            Humidity
30%
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 10,
          "height": "auto",
          "width": "100%",
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Last Watering:
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#128eb7",
              "borderColor": "#ddd",
              "borderRadius": 2,
              "borderWidth": 1,
              "elevation": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 35,
                "margin": 8,
                "padding": 4,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 25,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
               
              20 May 2018
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 35,
                "margin": 8,
                "padding": 4,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 25,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
               
              20:56
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 10,
          "height": "auto",
          "width": "100%",
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Last updated:
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#128eb7",
              "borderColor": "#ddd",
              "borderRadius": 2,
              "borderWidth": 1,
              "elevation": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 35,
                "margin": 8,
                "padding": 4,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 25,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
               
              24 October 2008
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 35,
                "margin": 8,
                "padding": 4,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 25,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
               
              12:02
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#31ce12",
        "borderRadius": 4,
        "marginBottom": 4,
        "marginTop": 8,
        "opacity": 1,
        "paddingVertical": 15,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
       
      Start Watering
    </Text>
  </View>
</View>
`;
